### PR TITLE
Updated background invocation `&` operator to use -WorkingDirectory when starting a new job

### LINKED ISSUE #10515: `&` background invocation operator should `-WorkingDirectory` when starting a job
PR #10324 is merged and now we can use `-WorkingDirectory` when starting a job for background invocation operator. 

The following code can be replaced:

https://github.com/PowerShell/PowerShell/blob/f69f30ba49c13ddd601724b35634bf4d1f44e31d/src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs#L529-L532

# Steps to reproduce

```powershell

```

# Expected behavior

```none

```

# Actual behavior

```none

```

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.0.0-daily.20190906
PSEdition                      Core
GitCommitId                    7.0.0-daily.20190906
OS                             Microsoft Windows 10.0.18978
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

### PR DESCRIPTION
**Pull request by ayousuf23**
# PR Summary
Updated the background invocation `&` operator to use -WorkingDirectory parameter when starting a new job. 

Fix #10515

## PR Context

Previously, `Set-Location` was appended to the script block of `Start-Job` in order to set the working directory of a new job invoked by the background invocation operator `&`. Now that `Start-Job` accepts a `-WorkingDirectory` parameter that sets the working directory of the new job, it makes sense to update the code for using the background invocation operator to use the new way of setting the working directory of a new job.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [] N/A or can only be tested interactively
    - **OR**
    - [ x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by iSazonov** [comment]
@ayousuf23 Please add test(s) for a scenario you enhance.

**Comment by anamnavi** [comment]
@ayousuf23 thanks for creating this PR, I like that you've added a thorough summary and context for this PR as this is really helpful :) the code looks good but having tests will just help us verify it works as expected, this is something we can talk about in the Semester of Code office hours this week, thanks again for creating this!

**Comment by iSazonov** [comment]
> Right now, the build is failing on my computer. I think this is because of other files (not the ones I changed)

Please create new branch on a commit before #15177, apply your commits and check if test fails still exist.

/cc @daxian-dbw

**Comment by ayousuf23** [comment]
@iSazonov Thank you for your suggestion. I will add another test.

**Comment by ayousuf23** [comment]
@iSazonov I pushed a commit that addresses your suggestions

File: src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs [hunk 1]
```diff
@@ -568,14 +564,22 @@ private static string GetParameterText(string parameterName)
                 var sb = ScriptBlock.Create(updatedScriptblock.ToString());
                 var commandInfo = new CmdletInfo("Start-Job", typeof(StartJobCommand));
                 commandProcessor = context.CommandDiscovery.LookupCommandProcessor(commandInfo, CommandOrigin.Internal, false, context.EngineSessionState);
-                var parameter = CommandParameterInternal.CreateParameterWithArgument(
+                var workingDirectoryParameter = CommandParameterInternal.CreateParameterWithArgument(
+                    parameterAst: pipelineAst,
```
Comment (anamnavi) [inline src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs:568]: perhaps make all the arguments passed in named parameters then (i.e `argumentName: argumentValue`)

**Review by anamnavi — APPROVED** [review]

**Review by iSazonov — APPROVED** [review]

**Comment by ayousuf23** [comment]
@anamnavi Sorry for the late response. I added parameter names for the arguments, as you suggested.

**Comment by ayousuf23** [comment]
@anamnavi @iSazonov Should I combine all the commits into one commit?

**Comment by iSazonov** [comment]
> @anamnavi @iSazonov Should I combine all the commits into one commit?

No, we will do squash in merge time.

**Comment by ayousuf23** [comment]
@iSazonov Is there anything remaining for me to do in order to get the PR merged?

**Comment by iSazonov** [comment]
@ayousuf23 We are waiting a review from anybody from MSFT team..

**Comment by ayousuf23** [comment]
@anamnavi Can you please review this PR?

**Comment by daxian-dbw** [comment]
@ayousuf23 Sorry for the delay of response from the team. Can you please mention the number of the issue that you fixed with this PR in the PR description (in the form `Fix #xxxxx`)? So the issue can be closed automatically when merging this PR.

**Comment by ayousuf23** [comment]
@daxian-dbw Thanks for your comment! I updated the PR's description.

**Comment by iSazonov** [comment]
@daxian-dbw I see the error on Windows CI "##[error]This is a scheduled windows-2016 brownout. The windows-2016 environment is deprecated and will be removed on March 15, 2022. For more details, see https://github.com/actions/virtual-environments/issues/4312"

image

**Review by TravisEz13 — APPROVED** [review]

**Comment by iSazonov** [comment]
@ayousuf23 Thanks for your contribution!

**Pull request merged**